deploy/eck-resources/examples/elasticsearch/detailed.yaml

```diff
@@ -0,0 +1,458 @@
+          #   whenUnsatisfiable: DoNotSchedule
+          #   labelSelector:
+          #     matchLabels:
+          #       elasticsearch.k8s.elastic.co/cluster-name: elasticsearch-sample
+          #       elasticsearch.k8s.elastic.co/statefulset-name: elasticsearch-sample-es-default
+
+          # Privilege and access control settings for the Elasticsearch Pods.
+          # ref: https://www.elastic.co/guide/en/cloud-on-k8s/2.2/k8s-security-context.html
+          #
+          securityContext: {}
+            # fsGroup: 0
+            # runAsUser: 1000
+            # runAsGroup: 0
+
+      # Volume Claim settings.
+      # ref: https://www.elastic.co/guide/en/cloud-on-k8s/2.2/k8s-volume-claim-templates.html
+      #
+      volumeClaimTemplates:
+      - metadata:
+          # The name of this volume must always be 'elasticsearch-data'.
+          #
+          name: elasticsearch-data
+        spec:
+          accessModes:
+          - ReadWriteOnce
+          resources:
+            requests:
+              storage: 100Gi
+          storageClassName: standard
+    - name: hot-node
+      config:
+        node.roles: ["data_hot", "ingest", "data_content"]
+        node.store.allow_mmap: false
+        transport.compress: true
+        # xpack.http.ssl.verification_mode: none
+        # xpack.security.audit.enabled: true
+        # xpack.security.authc.token.enabled: true
+        #
+        # Controlling of advanced zone awareness settings within Kubernetes, which allow you to
+        # create an availability zone-aware Elasticsearch cluster.
+        # ref: https://www.elastic.co/guide/en/cloud-on-k8s/current/k8s-advanced-node-scheduling.html#k8s-availability-zone-awareness
+        #
+        # node.attr.zone: ${ZONE}
+        # cluster.routing.allocation.awareness.attributes: k8s_node_name,zone
+      count: 3
+      podTemplate:
+        # Setting of custom labels and annotations on pods created by ECK.
+        # ref: https://www.elastic.co/guide/en/cloud-on-k8s/latest/k8s-customize-pods.html
+        metadata: {}
+          # labels:
+          #   my.custom.domain/label: "label-value"
+          # annotations:
+          #   my.custom.domain/annotation: "annotation-value"
+        spec:
+          containers:
+          # Settings related to the default 'elasticsearch' container.
+          #
+          - name: elasticsearch
+            resources:
+              requests:
+                cpu: 2
+                memory: 8Gi
+              limits:
+                cpu: 4
+                memory: 16Gi
+            # readinessProbe:
+            #   exec:
+            #     command:
+            #     - bash
+            #     - -c
+            #     - /mnt/elastic-internal/scripts/readiness-probe-script.sh
+            #   failureThreshold: 3
+            #   initialDelaySeconds: 10
+            #   periodSeconds: 5
+            #   successThreshold: 1
+            #   timeoutSeconds: 5
```

Comment:
Maybe you could use [YAML Anchor](https://helm.sh/docs/chart_template_guide/yaml_techniques/#yaml-anchors) to avoid duplication for all these kinds of configurations that may be reused in multiple parts of the values.

Reply:
I did initially try to use yaml anchors, and ran into helm issues with it, will try again though.

Reply:
>I did initially try to use yaml anchors, and ran into helm issues with it, will try again though.

TBH I never used them myself, but mostly saw them in other charts and PRs. It seems convenient in some use cases with YAML duplications like in umbrella charts or ECK spec but don't have more personal feedback on it.